main.go

```diff
@@ -31,6 +31,13 @@ var version string
 
 func main() {
 
+	product := lib.GetProductById(parameters.Product)
+	if product == nil {
+		fmt.Printf("Invalid product specified")
```

Comment:
Trailing `f` in `fmt.Printf` seems redundant.
Also maybe print out available values right away to ease user's experience?

Reply:
This is mostly why I used UsageMessage afterwards. Though, this functionality should _never_ be called (at least at the moment) because GetParameters already obtains the product and verifies it.(see https://github.com/warrensbox/terraform-switcher/pull/435#discussion_r1615273953).

But in this case, do you think saying `Invalid product specified by "-q"` and printing out usage would be good enough, since the usage provides the available parameters?
I wonder if there's a way to print usage, but mark as an argument as having an invalid value, so that the usage tells the user this.

Reply:
> But in this case, do you think saying `Invalid product specified by "-q"` and printing out usage would be good enough, since the usage provides the available parameters?

Yep, agree, should be sufficient to simplify code (and is a common approach I guess).